crates/router/src/core/webhooks/incoming.rs

```diff
@@ -1716,3 +1741,135 @@ async fn fetch_optional_mca_and_connector(
         Ok((None, connector, connector_name))
     }
 }
+
+async fn update_connector_mandate_details(
+    state: &SessionState,
+    merchant_account: &domain::MerchantAccount,
+    key_store: &domain::MerchantKeyStore,
+    connector_mandate_details: webhooks::ConnectorMandateDetails,
+    object_ref_id: api::ObjectReferenceId,
+    source_verified: bool,
+    event_type: webhooks::IncomingWebhookEvent,
+) -> CustomResult<(), errors::ApiErrorResponse> {
+    let payment_attempt =
+        get_payment_attempt_from_object_reference_id(state, object_ref_id, merchant_account)
+            .await?;
+    if should_update_connector_mandate_details(&payment_attempt, source_verified, event_type) {
+        if let Some(ref payment_method_id) = payment_attempt.payment_method_id {
+            let key_manager_state = &state.into();
+            let payment_method_info = state
+                .store
+                .find_payment_method(
+                    key_manager_state,
+                    key_store,
+                    payment_method_id,
+                    merchant_account.storage_scheme,
+                )
+                .await
+                .to_not_found_response(errors::ApiErrorResponse::PaymentMethodNotFound)?;
+
+            if payment_method_info.connector_mandate_details.is_none() {
```

Comment:
`is_none` won't work here, we should check the existence of mca_id in the hashmap